gossip/src/gossip_service.rs

```diff
@@ -193,37 +197,40 @@ pub fn discover(
     ))
 }
 
-/// Creates a ThinClient by selecting a valid node at random
+/// Creates a TpuClient by selecting a valid node at random
 pub fn get_client(
     nodes: &[ContactInfo],
-    socket_addr_space: &SocketAddrSpace,
     connection_cache: Arc<ConnectionCache>,
-) -> ThinClient {
-    let protocol = connection_cache.protocol();
-    let nodes: Vec<_> = nodes
-        .iter()
-        .filter_map(|node| node.valid_client_facing_addr(protocol, socket_addr_space))
-        .collect();
+) -> TpuClientWrapper {
     let select = thread_rng().gen_range(0..nodes.len());
-    let (rpc, tpu) = nodes[select];
-    ThinClient::new(rpc, tpu, connection_cache)
-}
 
-pub fn get_multi_client(
-    nodes: &[ContactInfo],
-    socket_addr_space: &SocketAddrSpace,
-    connection_cache: Arc<ConnectionCache>,
-) -> (ThinClient, usize) {
-    let protocol = connection_cache.protocol();
-    let (rpc_addrs, tpu_addrs): (Vec<_>, Vec<_>) = nodes
-        .iter()
-        .filter_map(|node| node.valid_client_facing_addr(protocol, socket_addr_space))
-        .unzip();
-    let num_nodes = tpu_addrs.len();
-    (
-        ThinClient::new_from_addrs(rpc_addrs, tpu_addrs, connection_cache),
-        num_nodes,
-    )
```

Comment:
this seems to be spinning up rpc multiple clients under the wrapper of a `ThinClient`. I was able to get similar performance with a single TPU Quic client. This similar performance was why changed it to a single TPU Client. However, I am not sure I have full context as to whether this needs to spin up multiple TPU clients at once??